.github/workflows/test.yml

```diff
@@ -13,52 +13,50 @@ on:
     - cron: '30 5 * * 1'
 
 jobs:
-  lint:
-    # It is possible for lints to break after merge (e.g., new clippy checks), but it's not worth
-    # breaking the build if that happens. So we'll skip this job for scheduled runs.
-    if: github.event_name != 'schedule'
-    runs-on: ubuntu-latest
+  test-matrix:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+    runs-on: ${{ matrix.os }}
     steps:
       - name: Checkout the repo
-        uses: actions/checkout@v3
+        uses: actions/checkout@master
 
       - name: Install Rust toolchain
         uses: actions-rs/toolchain@v1
         with:
           toolchain: stable
+          profile: minimal
+          components: rustfmt, clippy
+
+      - uses: Swatinem/rust-cache@v1
```

Comment:
I considered adding a special key here to address [this concern](https://github.com/phylum-dev/cli/issues/337#issuecomment-1116270098) expressed by @cd-work, but after reading the [details](https://github.com/Swatinem/rust-cache#cache-details) for this Github action, it looks like it is already using a very specific key (even including a hash of `Cargo.lock`), so I don't think it is necessary to add any additional keying until/unless we see problems.

Reply:
We should apply this change elsewhere if we can.